src/xenia/emulator.cc

```diff
@@ -285,6 +288,112 @@ X_STATUS Emulator::LaunchStfsContainer(std::wstring path) {
   return CompleteLaunch(path, "game:\\default.xex");
 }
 
+void Emulator::Pause() {
+  auto lock = global_critical_region::AcquireDirect();
+  if (paused_) {
+    return;
+  }
+  paused_ = true;
+
+  auto threads =
+      kernel_state()->object_table()->GetObjectsByType<kernel::XThread>(
+          kernel::XObject::kTypeThread);
+  for (auto thread : threads) {
+    if (!thread->can_debugger_suspend()) {
+      // Don't pause host threads.
```

Comment:
note that there are a few threads that are considered host threads that we may want to pause - like the vblank one, audio, etc - as they can make callbacks/interrupts into guest code. Maybe we need to tag the threads with a 'kSuspendOnPause' or something instead of using whether it's host or guest.